### DISCUSSION TITLE
Trigger Execution Id

### DISCUSSION THREAD

**Discussion by davidhjones** [quartznet/quartznet#952] [General] [answered]
Is there a mechanism for me to create / retrieve a unique identifier for the execution of a Trigger+Job, which is the same for a given execution of a trigger+job (regardless of if the trigger needs to retry, is executed on a different "host")?

Imagine I have Job A, which is on a simple recurring (every 10 minutes) trigger, Trigger 1.

The first time Trigger 1 Executed Job A, I'd like to establish an Execution Id (`IJobExecutionContext.FireInstanceId` won't suffice) -- lets call it **Execution A.1.1**. Imagine **Execution A.1.1** fails (`Execute` does not complete). Since my job `.RequestRecovery`, I'll receive a retry attempt at some point in the future. That happens, and the `context.FireInstanceId` new (different from the initial Execution), yet, I'd like to be able to retrieve the Execution Id of **Execution A.1.1** in order to "recover" from the failure (e.g. use this execution Id to look up the Execution State from a database). 

Is there any mechanism in Quartz that already exposes this information, or something I can leverage to store it?

**Comment by davidhjones** [comment, marked as answer]
I think I was mistaken here. I should be able to use the TriggerDataMap, but I has miss-registered my Job as a singleton, and thus my datamap was being overwritten on subsequent Executions.

**Reply by lahma** [reply]
Hey great that you got that sorted. Is it an option to use the official MS DI integration package? It should drive towards correct configuration.
